submissions/2024-12-12-OpenFF-Sage-2.0.0-Training-Optimization-Dataset-v1.0/generate-combined-dataset.py

```diff
@@ -0,0 +1,147 @@
+import logging
+logging.getLogger("openff").setLevel(logging.ERROR)
+
+filtered_and_combined = OptimizationResultCollection.parse_file(
+    "/Users/jennifer.clark/bin/openff-sage/data-set-curation/quantum-chemical/data-sets/1-2-0-opt-set-v3.json",
+#    "/Users/jennifer.clark/bin/openff-sage/data-set-curation/quantum-chemical/data-sets/1-2-0-td-set.json",
+)
+rec_and_mol = filtered_and_combined.to_records()
+
+print('Number of results: ',filtered_and_combined.n_results,flush=True)
+print('Finished converting to records',flush = True)
+
+# Need to input initial molecules so that QCA will recognize the inputs as the same.
+initial_mols = [rec[0].initial_molecule for rec in rec_and_mol]
+
+dataset_factory1 = OptimizationDatasetFactory()
+provenance1 = dataset_factory1.provenance(ToolkitRegistry([OpenEyeToolkitWrapper]))
+
+dataset1 = OptimizationDataset(
+    dataset_name="OpenFF Sage 2.0.0 Training Optimization v1.0",
+    dataset_tagline="B3LYP-D3BJ/DZVP conformers applicable to drug-like molecules for OpenFF 2.0.0 Sage",
+    description=("A quantum chemical (QC) dataset curated to train OpenFF 2.0.0 Sage, "
+    "with reparametrized Lennard-Jones (LJ) and valence parameters, the latter "
+    "relevent to this dataset. This QC dataset with the OpenFF default level of "
+    "theory, B3LYP-D3BJ/DZVP, is used to benchmark Sage geometries and energetics. "
+    "These optimized conformer geometries where used in conjunction with the QC "
+    "dataset used to train one dimensional torsional profiles. This Generation 2 "
+    "dataset increases chemical diversity when compared to Generation 1, which are "
+    "of value to our industry partners. Large molecules (>20 heavy atoms) were also "
+    "included, including more flexible molecules and a greater degree of conformational "
+    "variation which provide intramolecular interactions."),
+    provenance=provenance1,
+)
+dataset1.metadata.submitter = "Jennifer A Clark"
+dataset1.metadata.long_description_url = (
+        "https://github.com/openforcefield/qca-dataset-submission/tree/master/"
+        "submissions/"
+        "2024-12-12-OpenFF-Sage-2.0.0-Training-Optimization-Dataset-v1.0"
+    )
+
+
+# Have to add records this way to avoid a round trip through the toolkit.
+records_by_cmiles= {}
+for record, molecule in rec_and_mol:
+    cmiles = molecule.to_smiles(isomeric=True, explicit_hydrogens=True, mapped=True)
+    if cmiles in records_by_cmiles.keys():
+        records_by_cmiles[cmiles].append((record, molecule))
+    else:
+        records_by_cmiles[cmiles]=[(record, molecule)]
+
+for records in records_by_cmiles.values():
+    base_record, base_molecule = records[0]
+    base_molecule._conformers = [m.conformers[0] for _, m in records]
+
+    dataset1.add_molecule(
+        index=base_molecule.to_smiles(
+            isomeric=True, explicit_hydrogens=False, mapped=False
+        ),
+        molecule=None,
+        initial_molecules=[rec.initial_molecule for rec, _ in records],
```

Comment:
(not a review) Just for your future reference, this is the key difference forcing the use of the existing (`qcelemental`) molecules. The default usage here is to provide an `openff.toolkit.Molecule` to the `molecule` kwarg, which does the round-trip through the toolkit  Lexie mentioned in her comments. Providing `molecule=None` and `initial_molecules` (which are `qcelemental.models.Molecule`s) instead bypasses the round trip.

Reply:
Thanks Brent, helpful to know!